### PR TITLE
save form and case attachments to blob db

### PR DESCRIPTION
**Pull request by snopoke**
Mostly copied from `BlobMixin`
@millerdev 

buddy @biyeun

### CODE REVIEW

File: corehq/form_processor/models.py [hunk 1]
```diff
@@ -320,18 +328,41 @@ class AbstractAttachment(DisabledDbMixin, models.Model):
...
     def read_content(self):
-        with open(self.filepath, 'r+') as f:
-            content = f.read()
-        return content
+        db = get_blob_db()
+        try:
+            blob = db.get(self.name, self._blobdb_bucket())
+        except (KeyError, NotFound):
+            raise AttachmentNotFound(u"{model} attachment: {name!r}".format(
+                                   model=type(self).__name__, name=self.name))
```
Comment (dimagimon) [inline corehq/form_processor/models.py]: E126 continuation line over-indented for hanging indent

**Comment by czue** [comment]
build fail real, i think.
```
======================================================================
ERROR: test_basic_completed_by_user (corehq.apps.reports.tests.test_esaccessors.TestFormESAccessorsSQL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/apps/reports/tests/test_esaccessors.py", line 86, in test_basic_completed_by_user
    self._send_form_to_es(completion_time=datetime(2013, 7, 2))
  File "/mnt/corehq/apps/reports/tests/test_esaccessors.py", line 73, in _send_form_to_es
    form_pair = make_es_ready_form(metadata)
  File "/mnt/corehq/util/test_utils.py", line 315, in make_es_ready_form
    wrapped_form = get_form_ready_to_save(metadata, is_db_test=is_db_test)
  File "/mnt/corehq/util/test_utils.py", line 305, in get_form_ready_to_save
    Attachment(name='form.xml', raw_content=form_xml, content_type='text/xml')
  File "/mnt/corehq/form_processor/interfaces/processor.py", line 96, in store_attachments
    return self.processor.store_attachments(xform, attachments)
  File "/mnt/corehq/form_processor/backends/sql/processor.py", line 30, in store_attachments
    md5=attachment.md5,
  File "/mnt/corehq/form_processor/models.py", line 66, in md5
    return hashlib.md5(self.content).hexdigest()
TypeError: must be string or buffer, not instance
----------------------------------------------------------------------
```

File: corehq/form_processor/models.py [hunk 2]
```diff
@@ -319,19 +329,46 @@ class AbstractAttachment(DisabledDbMixin, models.Model):
...
     def read_content(self):
-        with open(self.filepath, 'r+') as f:
-            content = f.read()
-        return content
+        db = get_blob_db()
+        try:
+            blob = db.get(self.blob_id, self._blobdb_bucket())
+        except (KeyError, NotFound):
+            raise AttachmentNotFound(u"{model} attachment: {name!r}".format(
+                                   model=type(self).__name__, name=self.name))
```
Comment (dimagimon) [inline corehq/form_processor/models.py]: E126 continuation line over-indented for hanging indent

**Comment by dimagimon** [comment]
There are 20 errors:

* corehq/blobs/abstractdb.py, line 4 - F401 'base64' imported but unused
* corehq/blobs/abstractdb.py, line 5 - F401 'errno' imported but unused
* corehq/blobs/abstractdb.py, line 6 - F401 'os' imported but unused
* corehq/blobs/abstractdb.py, line 8 - F401 'shutil' imported but unused
* corehq/blobs/abstractdb.py, line 9 - F401 'sys' imported but unused
* corehq/blobs/abstractdb.py, line 11 - F401 'md5' imported but unused
* corehq/blobs/abstractdb.py, line 12 - F401 'commonprefix' imported but unused
* corehq/blobs/abstractdb.py, line 12 - F401 'dirname' imported but unused
* corehq/blobs/abstractdb.py, line 12 - F401 'exists' imported but unused
* corehq/blobs/abstractdb.py, line 12 - F401 'isabs' imported but unused
* corehq/blobs/abstractdb.py, line 12 - F401 'isdir' imported but unused
* corehq/blobs/abstractdb.py, line 12 - F401 'join' imported but unused
* corehq/blobs/abstractdb.py, line 12 - F401 'realpath' imported but unused
* corehq/blobs/abstractdb.py, line 12 - F401 'sep' imported but unused
* corehq/blobs/abstractdb.py, line 13 - F401 'uuid4' imported but unused
* corehq/blobs/abstractdb.py, line 15 - F401 'abstract' imported but unused
* corehq/blobs/abstractdb.py, line 17 - F401 'BlobInfo' imported but unused
* corehq/blobs/abstractdb.py, line 18 - F401 'BadName' imported but unused
* corehq/blobs/abstractdb.py, line 18 - F401 'NotFound' imported but unused
* corehq/form_processor/models.py, line 355 - E126 continuation line over-indented for hanging indent

**Pull request closed without merging**